application/src/main/java/org/togetherjava/tjbot/commands/moderation/NoteCommand.java

```diff
@@ -0,0 +1,106 @@
+     */
+    public NoteCommand(@NotNull ModerationActionsStore actionsStore, @NotNull Config config) {
+        super("note", "Writes a note about the given user", SlashCommandVisibility.GUILD);
+
+        getData()
+            .addOption(OptionType.USER, USER_OPTION, "The user who you want to write a note about",
+                    true)
+            .addOption(OptionType.STRING, CONTENT_OPTION,
+                    "The content of the note you want to write", true);
+
+        hasRequiredRole = Pattern.compile(config.getSoftModerationRolePattern()).asMatchPredicate();
+        this.actionsStore = Objects.requireNonNull(actionsStore);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption = event.getOption(USER_OPTION);
+        Member author = event.getMember();
+        Guild guild = event.getGuild();
+        String content = event.getOption(CONTENT_OPTION).getAsString();
+
+        if (!handleChecks(guild.getSelfMember(), author, targetOption.getAsMember(), content,
+                event)) {
+            return;
+        }
+
+        sendNote(targetOption.getAsUser(), author, content, guild, event);
+    }
+
+    @SuppressWarnings("BooleanMethodNameMustStartWithQuestion")
+    private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
+            @Nullable Member target, CharSequence content, @NotNull Interaction event) {
+        if (target != null && !ModerationUtils.handleCanInteractWithTarget(ACTION_VERB, bot, author,
+                target, event)) {
+            return false;
+        }
+        if (!ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event)) {
+            return false;
+        }
+        return ModerationUtils.handleReason(content, event);
+    }
+
+    private void sendNote(@NotNull User target, @NotNull Member author, @NotNull String content,
+            @NotNull ISnowflake guild, @NotNull Interaction event) {
+        storeNote(target, author, content, guild);
+        sendFeedback(target, author, content, event);
+    }
+
+    private void storeNote(@NotNull User target, @NotNull Member author, @NotNull String content,
+            @NotNull ISnowflake guild) {
+        logger.info("'{}' ({}) wrote a note about the user '{}' ({}) with content '{}'.",
+                author.getUser().getAsTag(), author.getId(), target.getAsTag(), target.getId(),
+                content);
+
+        actionsStore.addAction(guild.getIdLong(), author.getIdLong(), target.getIdLong(),
+                ModerationAction.NOTE, null, content);
+    }
+
+    private static void sendFeedback(@NotNull User target, @NotNull Member author,
+            @NotNull String noteContent, @NotNull Interaction event) {
+        MessageEmbed feedback = ModerationUtils.createActionResponse(author.getUser(),
+                ModerationAction.NOTE, target, null, noteContent);
+        event.replyEmbeds(feedback).queue();
```

Comment:
```suggestion
        MessageEmbed feedback = ModerationUtils.createActionResponse(author.getUser(),
                ModerationAction.NOTE, target, null, noteContent);
                
        event.replyEmbeds(feedback).queue();
```

To elaborate on the logic.

```java
storeNote(target, author, content, guild);
sendFeedback(target, author, content, event);
```

Here both method calls are 1 line, and it's clear.
In this case, one method call takes 2 lines, and it's harder to read because of that.

So having that gap, is nicer for readability imo, it makes it clear where the separation is.

Reply:
fair